modules/imgcodecs/test/test_tiff.cpp

```diff
@@ -117,6 +117,86 @@ TEST(Imgcodecs_Tiff, decode_tile_remainder)
     // What about 32, 64 bit?
 }
 
+TEST(Imgcodecs_Tiff, decode_12)
+{
+    /* see issue #21700
+    */
+    const string root = cvtest::TS::ptr()->get_data_path();
+    cv::Mat img8UC1 = imread(root + "readwrite/pattern_8uc1.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img8UC3 = imread(root + "readwrite/pattern_8uc3.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img8UC4 = imread(root + "readwrite/pattern_8uc4.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img10UC1 = imread(root + "readwrite/pattern_10uc1.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img10UC3 = imread(root + "readwrite/pattern_10uc3.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img10UC4 = imread(root + "readwrite/pattern_10uc4.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img12UC1 = imread(root + "readwrite/pattern_12uc1.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img12UC3 = imread(root + "readwrite/pattern_12uc3.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img12UC4 = imread(root + "readwrite/pattern_12uc4.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img14UC1 = imread(root + "readwrite/pattern_14uc1.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img14UC3 = imread(root + "readwrite/pattern_14uc3.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img14UC4 = imread(root + "readwrite/pattern_14uc4.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img16UC1 = imread(root + "readwrite/pattern_16uc1.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img16UC3 = imread(root + "readwrite/pattern_16uc3.tif", cv::IMREAD_UNCHANGED);
+    cv::Mat img16UC4 = imread(root + "readwrite/pattern_16uc4.tif", cv::IMREAD_UNCHANGED);
+    ASSERT_TRUE(img8UC1.size() == img10UC1.size());
+    ASSERT_TRUE(img8UC3.size() == img10UC3.size());
+    ASSERT_TRUE(img8UC4.size() == img10UC4.size());
+    ASSERT_TRUE(img8UC1.size() == img12UC1.size());
+    ASSERT_TRUE(img8UC3.size() == img12UC3.size());
+    ASSERT_TRUE(img8UC4.size() == img12UC4.size());
+    ASSERT_TRUE(img8UC1.size() == img14UC1.size());
+    ASSERT_TRUE(img8UC3.size() == img14UC3.size());
+    ASSERT_TRUE(img8UC4.size() == img14UC4.size());
+    ASSERT_TRUE(img8UC1.size() == img16UC1.size());
+    ASSERT_TRUE(img8UC3.size() == img16UC3.size());
+    ASSERT_TRUE(img8UC4.size() == img16UC4.size());
+
+    const double maxDiff = 256;//samples do not have the exact same values because of the tool that created them
+    cv::Mat tmp;
+    double diff = 0;
+
+    if (!img16UC1.empty())
```

Comment:
`.empty()` is a test failure.

Reply:
I was unsure about what to do, because the test will fail on the buildbot as long as the sample files [#962](https://github.com/opencv/opencv_extra/pull/962) are not commited